packages/amplify-category-function/src/provider-utils/awscloudformation/utils/buildLayer.ts

```diff
@@ -0,0 +1,52 @@
+import { $TSContext, pathManager } from 'amplify-cli-core';
+import { FunctionRuntimeLifecycleManager, BuildRequest, BuildType } from 'amplify-function-plugin-interface';
+import * as path from 'path';
+import { categoryName } from '../../../constants';
+import { BuildRequestMeta } from './buildFunction';
+import { ServiceName } from './constants';
+import { LayerConfiguration, loadLayerConfigurationFile } from './layerConfiguration';
+
+export const buildLayer = async (context: $TSContext, { resourceName, lastBuildTimestamp }: BuildRequestMeta) => {
+  const layerConfig: LayerConfiguration = loadLayerConfigurationFile(resourceName);
+  const resourcePath = path.join(pathManager.getBackendDirPath(), categoryName, resourceName);
+
+  const layerCodePath = path.join(resourcePath, 'lib', layerConfig.runtimes[0].layerExecutablePath);
+
+  const runtimePlugin: FunctionRuntimeLifecycleManager = (await context.amplify.loadRuntimePlugin(
+    context,
+    layerConfig.runtimes[0].runtimePluginId,
+  )) as FunctionRuntimeLifecycleManager;
+
+  const depCheck = await runtimePlugin.checkDependencies(layerConfig.runtimes[0].value);
+  if (!depCheck.hasRequiredDependencies) {
+    context.print.error(depCheck.errorMessage || `Required dependencies to build ${resourceName} are missing`);
+    const err = new Error(`Required dependencies to build ${resourceName} are missing`);
+    delete err.stack;
```

Comment:
Please use `err.stack = undefined`... `delete` is a somewhat expensive operation that we should avoid if possible.